### PR TITLE
Fix slicing decorator

### PR DESCRIPTION
**Pull request by vincentschen**
**Changes**
(_Note:_ Take 2 on https://github.com/HazyResearch/snorkel/pull/1381)
* Fixes problem where initializing with `@slicing_function()` would actually initialize a `LabelingFunction`

**Test Plan**
* passes tox

### CODE REVIEW

**Comment by vincentschen** [comment]
@henryre can you have another look here? added some tests + made some additional changes to simplify

File: snorkel/labeling/lf/nlp.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def _create_or_check_preprocessor(
             cls._nlp_config = SpacyPreprocessorConfig(nlp=nlp, parameters=parameters)
         elif parameters != cls._nlp_config.parameters:
             raise ValueError(
-                "{cls.__name__} already configured with different parameters: "
+                f"{cls.__name__} already configured with different parameters: "
```
Comment (henryre) [inline snorkel/labeling/lf/nlp.py]: nice catch!

File: snorkel/slicing/sf/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from .core import SlicingFunction, slicing_function  # noqa: F401
+from .nlp import NLPSlicingFunction, nlp_slicing_function  # noqa: F401
```
Comment (henryre) [inline snorkel/slicing/sf/__init__.py]: rm. this puts spacy on the main import path.

File: snorkel/slicing/sf/core.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+class SlicingFunction(LabelingFunction):
+    """Base class for slicing functions.
+
+    See ``snorkel.labeling.lf.LabelingFunction`` for details.
+    """
+
+    pass
+
+
+class slicing_function:
```
Comment (henryre) [inline snorkel/slicing/sf/core.py]: maybe add a quick TODO here indicating that we're planning on having a common parent decorator.

**Pull request merged**